src/nostr-post/nostr-post.ts

```
@@ -65,32 +105,153 @@ export default class NostrPost extends NostrEventComponent {
       }
     } catch (err) {
+      if (seq !== this.statsRequestSeq || this.event?.id !== eventId) {
+        return;
+      }
+
       const msg = err instanceof Error ? err.message : 'Failed to load post stats';
       console.error('[NostrPostComponent] ' + msg, err);
       this.eventStatus.set(NCStatus.Error, msg);
     } finally {
+      if (seq !== this.statsRequestSeq || this.event?.id !== eventId) {
+        return;
```

Biome (2.4.9)

[error] Unsafe usage of 'return'.

(lint/correctness/noUnsafeFinally)